src/onefetch/repo.rs

```diff
@@ -269,3 +282,9 @@ pub fn is_valid(repo_path: &str) -> Result<bool> {
     let repo = Repository::open_ext(repo_path, RepositoryOpenFlags::empty(), Vec::<&Path>::new());
     Ok(repo.is_ok() && !repo?.is_bare())
 }
+
+pub fn is_bot(author: Signature, bot_exclude_pattern: &Option<String>) -> bool {
+    let author_name = String::from_utf8_lossy(author.name_bytes()).into_owned();
+    let re = Regex::new(bot_exclude_pattern.as_ref().unwrap()).unwrap();
```

Comment:
Oh, also you might want to make use of `lazy_static!`, since I believe that this will compile regex on each line that `is_bot` is called on, otherwise.

Reply:
Great remark! I chose to use a `Option<Regex>` instead of a `Option<String>` for the bot_exclude_pattern so that I can provide an pre-compiled Regex pattern to the program.